source/core/memory.cpp

```diff
@@ -291,36 +292,33 @@ void Memory::deserializeJSON(const Json& json) {
   }
   // TODO::check types!
   // check that sizes match
-  std::size_t byteCount = *byteCountIt;
-  if (_byteCount != byteCount) {
-    std::string expected = std::to_string(_byteCount);
-    std::size_t byteCount = *byteCountIt;
-    std::string recieved = std::to_string(byteCount);
-    throw DeserializationError(
-        "Could not deserialize Memory: ByteCount did not match: expected \"" +
-        expected + "\" to be equal to \"" + recieved + "\"");
+  size_t byteCount = *byteCountIt;
+  if (_byteCount < byteCount) {
+    _byteCount = byteCount;
+    _data = MemoryValue(byteCount * _byteSize);
+    _sizeCallback(_byteCount);
   }
-  std::size_t byteSize = *byteSizeIt;
+  size_t byteSize = *byteSizeIt;
   if (_byteSize != byteSize) {
     std::string expected = std::to_string(_byteSize);
     std::string recieved = std::to_string(byteSize);
     throw DeserializationError(
         "Could not deserialize Memory: ByteSize did not match: expected \"" +
         expected + "\" to be equal to \"" + recieved + "\"");
   }
-  std::size_t lineLength = json[_lineLengthStringIdentifier].get<std::size_t>();
+  size_t lineLength = json[_lineLengthStringIdentifier].get<size_t>();
   if (lineLength <= 0) {
     throw DeserializationError("Could not deserialize Memory: lineLength == 0");
   }
 
   // clear the current memory to write the zeros in the serialized data
   clear();
```

Comment:
You could optimize the usage of clear and avoid clearing if you resize the memory